src/crimson/osd/ops_executer.h

```diff
@@ -75,7 +75,7 @@ class OpsExecuter {
   ObjectContextRef obc;
   PG& pg;
   PGBackend& backend;
-  Ref<MOSDOp> msg;
+  OSDOpParams osd_op_params;
```

Comment:
Do we need `osd_op_params` at this level? It could be in-place constructed by `submit_changes()`. All we need here is just the `user_modify` flag, I guess.

Constructing-on-the-fly would be helpful when transforming `submit_changes(task_always_executed)` into `commit(task_executed_when_txn_is_non_empty)` (there is a comment about the conditional execution) to save a bunch of cycles on the read path.